src/FuncSharp/Collections/ReadOnlyList.cs

```diff
@@ -1,158 +1,40 @@
-    {
-        return source is not null && source.Count == 1;
-    }
-
-    public static T Single<T>(this IReadOnlyList<T> source)
-    {
-        if (!source.IsSingle())
-        {
-            throw new ArgumentException("Source is not a single element.");
-        }
-
-        return source[0];
-    }
-
-    public static IOption<T> SingleOption<T>(this IReadOnlyList<T> source)
-    {
-        return source.IsSingle().Match(
-            t => Option.Valued(source[0]),
-            f => Option.Empty<T>()
-        );
-    }
-
-    public static T First<T>(this IReadOnlyList<T> source)
-    {
-        if (source.IsEmpty())
-        {
-            throw new ArgumentException("Source is empty.");
-        }
-
-        return source[0];
-    }
-
-    public static IOption<T> FirstOption<T>(this IReadOnlyList<T> source)
-    {
-        return source.IsEmpty().Match(
-            t => Option.Empty<T>(),
-            f => Option.Valued(source[0])
-        );
-    }
-
-    public static T Second<T>(this IReadOnlyList<T> source)
-    {
-        if (source.Count <= 1)
-        {
-            throw new ArgumentException("Source has less than 2 elements.");
-        }
-
-        return source[1];
-    }
-
-    public static T Last<T>(this IReadOnlyList<T> source)
-    {
-        if (source.Count == 0)
-        {
-            throw new ArgumentException("Source is empty.");
-        }
-
-        return source[source.Count - 1];
-    }
-
-    public static IOption<T> LastOption<T>(this IReadOnlyList<T> source)
-    {
-        return source.IsEmpty().Match(
-            t => Option.Empty<T>(),
-            f => Option.Valued(source[source.Count - 1])
-        );
-    }
-
-    public static T ElementAt<T>(this IReadOnlyList<T> source, int index)
-    {
-        return source[index];
-    }
-
-    public static IOption<T> ElementAtOption<T>(this IReadOnlyList<T> source, NonNegativeInt index)
-    {
-        return (source.Count > index).Match(
-            t => source[index].ToOption(),
-            f => Option.Empty<T>()
-        );
-    }
-
-    public static int IndexOf<T>(this IReadOnlyList<T> source, T item)
-    {
-        for (var i = 0; i < source.Count; i++)
-        {
-            if (source[i].SafeEquals(item))
-            {
-                return i;
-            }
-        }
-        return -1;
-    }
-}
-
 public static class ReadOnlyList
 {
     public static IReadOnlyList<T> Create<T>(params T[] values)
     {
         return CreateFlat(values);
     }
 
-    [Obsolete("Use CreateFlat instead. This method will be gone once Mews replaces all the usages.", true)]
-    public static IReadOnlyList<T> Create<T>(IEnumerable<T> values)
-    {
-        return values.ToList();
-    }
-
-    [Obsolete("Use CreateFlat instead. This method will be gone once Mews replaces all the usages.", true)]
-    public static IReadOnlyList<T> Create<T>(IReadOnlyList<T> values)
+    public static IReadOnlyList<T> Create<T>(T head, IEnumerable<T> tail)
```

Comment:
Why do we need to handle this case separately? Do we want to return `INonEmptyEnumerable` instead?